Deprecate SmurfNaming and superflus Interfaces

flow-control opened the pull request:
This pull request will deprecate the smurf naming in services and remove superflus interfaces where not needed:

- [x] deprecate `LegacyService` in favor of `Legacy`
- [x] deprecate `LegacyServiceInterface`
- [x] deprecate `KeyRegistryInterface`
- [x] deprecate `AuthenticationService` in favor of `Authentication`
- [x] deprecate `AuthenticationServiceInterface`
- [x] deprecate `AuthorizationService` in favor of `Authorization`
- [x] deprecate `AuthorizationServiceInterface`
- [x] deprecate `GraphQLQueryHandlerInterface`
- [x] deprecate `ErrorCodeProviderInterface`
- [x] deprecate `RequestReaderInterface`
- [x] deprecate `ResponseWriterInterface`
- [x] deprecate `SchemaFactoryInterface`
- [x] deprecate `BeforeAuthorizationEvent` in favor of `BeforeAuthorization`

alfredbez commented:
LGTM, although I was not sure about the empty `deprecate` tags like in https://github.com/OXID-eSales/graphql-base-module/pull/9/files#diff-5c34845d0322e2cb17a0073b4e0d066dR18

The build failed during the infection-step:
* You need to change the class-name in `infection.json.dist` like this:
```
diff --git a/infection.json.dist b/infection.json.dist
index aae271e..7b7e692 100644
--- a/infection.json.dist
+++ b/infection.json.dist
@@ -21,12 +21,14 @@
         "@arithmetic": {
             "ignore": [
                 "OxidEsales\\GraphQL\\Base\\Service\\AuthenticationService::getTokenBuilder",
+                "OxidEsales\\GraphQL\\Base\\Service\\Authentication::getTokenBuilder",
                 "OxidEsales\\GraphQL\\Base\\Exception\\*"
             ]
         },
         "@number": {
             "ignore": [
                 "OxidEsales\\GraphQL\\Base\\Service\\AuthenticationService::getTokenBuilder",
+                "OxidEsales\\GraphQL\\Base\\Service\\Authentication::getTokenBuilder",
                 "OxidEsales\\GraphQL\\Base\\Service\\KeyRegistry::generateSignatureKey"
             ]
         },
```
* I'm not 100% sure how to proceed with infection:
    * There are currently two escaped mutants (we do not check if a token has an issuer and an audience in the tests, not sure if this is something we want to check even)
    * Looks like all of the `QueryBuilder` related things in the filters are not tested in the Unit-Tests
    * It would be awesome to add integration tests to CI, too
    * Maybe we should lower the threshold for now

flow-control commented:
Updated `infection.json.dist` and fixed the pipeline :wink:

I played a bit with the escaped mutations, but those are very tricky to catch. OTOH we could maybe catch these if we refactor the token generation to its own factory. But this is out of scope for this PR
